src/components/DataProvider/index.js

```diff
@@ -36,20 +37,40 @@ export default class DataProvider extends Component {
   };
 
   async componentDidMount() {
-    await this.fetchData('MOUNTED');
+    await Promise.map(this.props.series, s => this.fetchData(s.id, 'MOUNTED'));
     if (this.props.updateInterval) {
       this.fetchInterval = setInterval(() => {
-        this.fetchData('INTERVAL');
+        Promise.map(this.props.series, s => this.fetchData(s.id, 'INTERVAL'));
       }, this.props.updateInterval);
     }
   }
 
+  async componentDidUpdate(prevProps, prevState) {
+    const { series: prevSeries } = prevState;
+    if (!prevSeries) {
```

Comment:
It seems like an issue here, since series is not defined in state, the if statement will be everytime `true` and the logic below will never be addressed.